jb/itch5/compute_book_cache_aware.hpp

```diff
@@ -0,0 +1,214 @@
+#ifndef jb_itch5_compute_book_cache_aware_hpp
+#define jb_itch5_compute_book_cache_aware_hpp
+
+#include <jb/itch5/add_order_message.hpp>
+#include <jb/itch5/add_order_mpid_message.hpp>
+#include <jb/itch5/order_book_cache_aware.hpp>
+#include <jb/itch5/order_cancel_message.hpp>
+#include <jb/itch5/order_delete_message.hpp>
+#include <jb/itch5/order_executed_message.hpp>
+#include <jb/itch5/order_executed_price_message.hpp>
+#include <jb/itch5/order_replace_message.hpp>
+#include <jb/itch5/stock_directory_message.hpp>
+#include <jb/itch5/unknown_message.hpp>
+
+#include <boost/functional/hash.hpp>
+#include <chrono>
+#include <functional>
+#include <unordered_map>
+
+namespace jb {
+namespace itch5 {
+
+/**
+ * An implementation of jb::message_handler_concept to compute the book depth.
+ *
+ * Keep a collection of all the order books, and forward the right
+ * updates to them as it handles the different message types in ITCH-5.0.
+ * Calls the callback on any event (changes to the book)
+ */
+
+class compute_book_cache_aware {
```

Comment:
This seems to be largely a cut&paste from jb::itch5::compute_book.  No problem with that, except that compute_book changed a lot since you cut&pasted.  The comments got more explicit, the callback is more flexible.  Please update to the newer version of compute_book.

Or better yet, refactor to a template based on the type of jb::itch5::order_book.